### ISSUE TITLE
Switch to AGPL-3.0?

### ISSUE THREAD

**Issue by ypid** [debops/debops#1286]
With the licensing situation of DebOps now defined (#1275) we could discuss the pro/contra of the AGPL-3.0.

In the past, I re-licensed code to GPL-3.0-only (https://github.com/debops/debops/commit/22412d27086a433d04cf5a98145f9666f7725048). Are the reasons from back then still a concern? From my memory, I think it was mixing of code.

Switching to AGPL-3.0 would be easy as no approval from contributors is required as far as I know (ref: https://www.gnu.org/licenses/gpl-faq.en.html#AllCompatibility).

I am not sure yet what specific variant of the AGPL: Either AGPL-3.0-only or AGPL-3.0-or-later.

**Comment by drybjed** [comment]
Keep in mind that not all contributors are accounted for in all files yet. I suppose that we could debate at what point a contribution warrants to be counted in the copyright list - small commits that fix English mistakes or update code indentation probably wouldn't count as a copyrightable. It would be interesting to look at other projects for guidelines and write them down in the `CONTRIBUTING.rst` documentation page.

AfferoGPL seems to be a bit controversial - some organizations like Google outright ban its use by the developers. This could be considered a disadvantage for the project in some way, which I would want to avoid. Although I'm not a lawyer, so don't quote me on that.

The current licens of DebOps aligns with the license of Ansible which is its backbone and skeleton. As far as I can tell, if somebody can use Ansible in their work, using DebOps with it should be straightforward due to the same license.

I'm not currently aware of the reach of the project among the Ansible community; I know that it is used on some universities either as a development tool or for small infrastructure projects, other people also use it in companies but I'm not sure what amount could or would want to declare that officially as being "based on" DebOps. If we are talking about changing the license, how would that affect our userbase?

We also need to consider the license compatibility - for example, Apache 2.0 code can be added to the GPLv3 code, but not the other way around. How would that change with a move to AGPL? Would we be unintentionally cutting ourselves from including some specific code needed for deployment?

**Comment by ypid** [comment]
The thing with Google seems to be correct. Still, when pushing for a world of only Free Software, the AGPL has advantages. And DebOps has network components so this is not irrelevant.

GPL and AGPL are for the most part identical. Even the compatibility matrix in the GPL FAQ tread them as identical in terms of comparability. I did not find infos on differences of the two GPL variants when it comes to the Apache license. The result when combined will be either GPL or AGPL licensed as a whole.

https://www.gnu.org/licenses/gpl-faq.en.html#AllCompatibility:

> Each place that the matrix states GPLv3, the same statement about compatibility is true for AGPLv3 as well.

Note that I also have not done comprehensive research now. But I would check this in more detail in case you are open about this direction. No time pressure. We can just keep this in the back of our head for a while and see if we spot any showstoppers.

**Comment by drybjed** [comment]
Agreed, this doesn't have to be solved right away. I would like to hear input from other community members interested in this topic. For me AGPL is not a hard "no", if enough people think that move to it would be benefical - why not. I just think that having the same license as Ansible itself is enough.
